### PR TITLE
updated and fixed typeform detectors

### PR DESCRIPTION
**Pull request by kashifkhan0771**
### Description:
This PR improve overall logic of TypeForm detectors and fixes it's integration test cases.
Screenshot from 2024-12-12 16-21-36

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/typeform/v2/typeform_integration_test.go [hunk 1]
```diff
@@ -86,40 +89,6 @@ func TestTypeform_FromChunk(t *testing.T) {
 			wantErr:             false,
 			wantVerificationErr: false,
 		},
-		{
-			name: "found, would be verified if not for timeout",
```
Comment (kashifkhan0771) [inline pkg/detectors/typeform/v2/typeform_integration_test.go:90]: Even with 1 Microsecond timeout this test case was failing with no error 😄 (🥲) So I removed them.

File: pkg/detectors/typeform/v2/typeform_integration_test.go [hunk 2]
```diff
@@ -1,6 +1,3 @@
-//go:build detectors
```
Comment (ahrav) [inline pkg/detectors/typeform/v2/typeform_integration_test.go:1]: question: Have we started removing the build tag from detector integration tests, or is it only being removed from non-integration tests? Sorry if i'm late to the part on this. 😅

**Review by ahrav — APPROVED** [review]
Single non-blocking comment, but otherwise LGTM! Thanks for getting this cleaned up. ❤️

File: pkg/detectors/typeform/v2/typeform_integration_test.go (hunk 2 above)
Comment (kashifkhan0771) [reply pkg/detectors/typeform/v2/typeform_integration_test.go:1]: Ah good catch 😃 I removed it accidentally

**Review by ahrav — APPROVED** [review]

**Pull request merged**